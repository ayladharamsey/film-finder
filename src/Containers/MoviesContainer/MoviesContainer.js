import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesContainer.scss';

const MoviesContainer = ({ movieData }) => {

  const movies = movieData.map(prop => {
    return <MovieCard 
      key={prop.id}
      id={prop.id} 
      movieRating={prop.movieRating}
      backgroundImage={prop.backgroundImage}
      overview={prop.overview}
      posterImage={prop.posterImage}
      title={prop.title}
      releaseDate={prop.releaseDate}
      isFavorited={prop.isFavorited}
      />
  })
  

  return (
    <section className="movies-container">
      <h1>
        This is Movies Container::
        {movies}
      </h1>
    </section>
  )
}

export default MoviesContainer;