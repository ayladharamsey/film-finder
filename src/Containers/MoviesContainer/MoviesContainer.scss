
* {
  margin: 0px; 
  padding: 0px; 
}

.scroll-wrapper {
  width: 70%;
  transform: rotate(-90deg) translateX(-100vh);
  transform-origin: top left;
  overflow-y: scroll;
  overflow-x: hidden;
  position: absolute;
  margin-top: 14%;
  height: 100vw;
}

.movies-container {
  padding: 10px;
  height: 100vh;
  display:flex;
  justify-content: center;
  flex-direction: row;
  width: fit-content;
  transform: rotate(90deg) translateY(-105vh);
  transform-origin: top left;
    & :hover {
      transition: all .2s ease-in-out;
      transform: scale(1.1)
  }  
}

