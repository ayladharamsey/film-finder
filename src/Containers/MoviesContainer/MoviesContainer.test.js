import React from 'react';
import { shallow } from 'enzyme';
import MoviesContainer from './MoviesContainer';

describe('MoviesContainer container tests', () => {

  it('MoviesContainer should match its snapshot', () => {
    const mockMovieData = [{id:1, title:'The Horror'}]
    const wrapper = shallow( <MoviesContainer movieData={mockMovieData} /> )
    expect(wrapper).toMatchSnapshot();
  })
})