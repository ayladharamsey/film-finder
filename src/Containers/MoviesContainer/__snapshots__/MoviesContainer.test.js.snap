// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer container tests MoviesContainer should match its snapshot 1`] = `ShallowWrapper {}`;
